apps/dashboard/src/components/auth/questionnaire-form.tsx

```diff
@@ -0,0 +1,199 @@
+import { useForm, Controller } from 'react-hook-form';
+import { updateClerkOrgMetadata } from '../../api/organization';
+import { hubspotCookie } from '../../utils/cookies';
+import { identifyUser } from '../../api/telemetry';
+import { useTelemetry } from '../../hooks';
+import { TelemetryEvent } from '../../utils/telemetry';
+import { useNavigate } from 'react-router-dom';
+import { ROUTES } from '../../utils/routes';
+
+interface QuestionnaireFormData {
+  jobTitle: JobTitleEnum;
+  organizationType: OrganizationTypeEnum;
+  companySize?: CompanySizeEnum;
+}
+
+export function QuestionnaireForm() {
+  const { control, watch, handleSubmit } = useForm<QuestionnaireFormData>();
+  const track = useTelemetry();
+  const navigate = useNavigate();
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const selectedJobTitle = watch('jobTitle');
+  const selectedOrgType = watch('organizationType');
+  const companySize = watch('companySize');
+
+  const shouldShowCompanySize =
+    (selectedOrgType === OrganizationTypeEnum.COMPANY || selectedOrgType === OrganizationTypeEnum.AGENCY) &&
+    !!selectedJobTitle;
+
+  const isFormValid = React.useMemo(() => {
+    if (!selectedJobTitle || !selectedOrgType) return false;
+    if (shouldShowCompanySize && !companySize) return false;
+
+    return true;
+  }, [selectedJobTitle, selectedOrgType, shouldShowCompanySize, companySize]);
+
+  const onSubmit = async (data: QuestionnaireFormData) => {
+    setIsSubmitting(true);
+
+    try {
+      await updateClerkOrgMetadata({
+        companySize: data.companySize,
+        jobTitle: data.jobTitle,
+        organizationType: data.organizationType,
+      });
+
+      const hubspotContext = hubspotCookie.get();
+
+      await identifyUser({
+        jobTitle: data.jobTitle,
+        pageUri: window.location.href,
+        pageName: 'Create Organization Form',
+        hubspotContext: hubspotContext || '',
+        companySize: data.companySize,
+        organizationType: data.organizationType,
+      });
+
+      track(TelemetryEvent.CREATE_ORGANIZATION_FORM_SUBMITTED, {
+        location: 'web',
+        jobTitle: data.jobTitle,
+        companySize: data.companySize,
+        organizationType: data.organizationType,
+      });
+
+      navigate(ROUTES.USECASE_SELECT);
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  return (
+    <>
+      <div className="w-[564px] px-0 py-[60px]">
+        <div className="flex flex-col items-center gap-8">
+          <div className="flex w-[350px] flex-col gap-1">
+            <div className="flex w-full items-center gap-1.5">
+              <div className="flex flex-1 flex-col gap-1">
+                <StepIndicator step={2} />
+                <CardTitle className="text-lg font-medium text-[#232529]">
+                  Help us personalize your experience
+                </CardTitle>
+              </div>
+            </div>
+            <CardDescription className="text-xs text-[#99A0AE]">
```

Comment:
All "gray" text colors are named `foreground` in our tw config, so `text-soft-400` is `text-foreground-400`.

Reply:
Yep perfect, just chatted with George on it 💪